pymtl3/dsl/Component.py

```diff
@@ -490,15 +506,17 @@ def unlock_simulation( s ):
     except:
       raise AttributeError("Cannot unlock an unlocked/never locked model.")
 
+    swapped_values  = defaultdict(list)
```

Comment:
Is this mostly a performance optimization that tries to remove unnecessary attribute lookups?

Reply:
Yes ... I try to only do it when it doesn't hurt readability ...